### PR TITLE
[PM-35945] Browser: Show the Health Overview after a scan (gauge and risk categories)

### PR DESCRIPTION
**Pull request by AlexRubik**
## 🎟️ Tracking

[PM-35945](https://bitwarden.atlassian.net/browse/PM-35945)

## 📔 Objective

Builds the Health Overview a premium user sees on the extension's Health tab after a scan: the At-Risk Gauge with its heading and count, plus the Exposed, Weak, and Reused categories with deduplicated counts.

Mostly composition, since everything it composes is already merged. The overview and a reusable category row are new components in `bitwarden_license/bit-browser/src/popup/dirt/health/`; the licensed shell imports the overview directly, replacing the `RESULTS PLACEHOLDER` from [#22209](https://github.com/bitwarden/clients/pull/22209). Only open-source file touched is `messages.json`.

Layout and copy match the Figma frame. Copy is still worth a content pass before merge, since locale keys are immutable once shipped.

### Notes for reviewers

| Topic | Detail |
|---|---|
| Interim scan trigger | The report service is compute-only, so the overview scans on init and takes only the first cipher emission, keeping the breach lookup off every vault edit. One `TODO` block for [PM-39223](https://bitwarden.atlassian.net/browse/PM-39223) to lift out; a test pins the scan-once behaviour. |
| Blank until the report resolves | A green gauge mid-scan would mislead. Errors are logged and nothing renders; the failure screen is [PM-39223](https://bitwarden.atlassian.net/browse/PM-39223)'s. |
| Rows link to `/health/{category}` | Matches the extension's top-level drill-down convention. Those routes land with [PM-35946](https://bitwarden.atlassian.net/browse/PM-35946), so taps log `NG04002` and do not navigate yet. |
| No premium gate | `healthEnabled$` checks the flag and org tier, not premium. That branch is [PM-35944](https://bitwarden.atlassian.net/browse/PM-35944)'s; flagged for QA. |
| `I18nPipe` over `JslibModule` | Matches the shell and gauge beside it. |

**Verification:** 61 unit tests, ESLint and `test:locales` clean, and 25 end-to-end checks in a real browser across both states, compact mode, and dark mode.

## 📸 Screenshots

1-at-risk
2-healthy
3-healthy-dark
4-compact-mode

[PM-35945]: https://bitwarden.atlassian.net/browse/PM-35945?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[PM-39223]: https://bitwarden.atlassian.net/browse/PM-39223?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: bitwarden_license/bit-browser/src/popup/dirt/health/health-overview.component.ts [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+   * Health tab open — the agreed cadence, with no caching and no manual rescan
+   * — and takes only the first cipher emission so an expensive breach lookup is
+   * not re-run on every vault change.
+   */
+  private readonly report = toSignal(
+    this.accountService.activeAccount$.pipe(
+      getUserId,
+      switchMap((userId) =>
+        this.cipherService.cipherViews$(userId).pipe(
+          take(1),
```
[the remark this answers is not in the document]
Comment (AlexRubik) [reply bitwarden_license/bit-browser/src/popup/dirt/health/health-overview.component.ts:117]: Already addressed, and with exactly the operator suggested here: `filterOutNullish()` before `take(1)`, fixed in b6b9b3eab2. This comment was written about 18 minutes before that commit was pushed, so it was reviewing the previous head.

Agreed on the severity reasoning too. The failure was silent rather than reaching `catchError`, which is what made it worth fixing before anything else.

Also added a regression test that seeds the stream with `null` and asserts no scan runs off it. I confirmed the test fails against the unfixed component before keeping it, so it genuinely pins the behaviour.

Resolving.

File: bitwarden_license/bit-browser/src/popup/dirt/health/health-overview.component.ts [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+  private readonly accountService = inject(AccountService);
+  private readonly cipherService = inject(CipherService);
+  private readonly vaultHealthReportService = inject(VaultHealthReportService);
+  private readonly logService = inject(LogService);
+
+  /**
+   * The latest scan's result, or null while it is still running or after it
+   * failed.
+   *
+   * Read through a single `toSignal`, so there is exactly one subscription no
```
Comment (lastbestdev) [inline bitwarden_license/bit-browser/src/popup/dirt/health/health-overview.component.ts]: Nit: a lot to read here and not a lot of value add IMO. Comments explaining how current code should not be rewritten differently in the future is over-prescriptive. I'm all for TODO comment placeholders, but the second paragraph here could be reduced to `TODO: add the scan trigger, loading and failure state in PM-39223`

File: bitwarden_license/bit-browser/src/popup/dirt/health/health-overview.component.ts [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+    this.accountService.activeAccount$.pipe(
+      getUserId,
+      switchMap((userId) =>
+        this.cipherService.cipherViews$(userId).pipe(
+          // cipherViews$ is cached with shareReplay and emits null when the
+          // decrypted ciphers are cleared (lock, logout, vault clear). Because
+          // the buffer is kept with refCount: false, a later subscriber can
+          // receive that null as its FIRST emission. Taking it would scan an
+          // empty vault and, since take(1) then completes, leave a permanent
+          // "Your vault is healthy" on a vault that was never actually read.
```
Comment (lastbestdev) [inline bitwarden_license/bit-browser/src/popup/dirt/health/health-overview.component.ts]: Nit: I think if we need to know about how `cipherViews$` is working, the developer can go to and read the definition. I think this would be fine without the comment.

File: bitwarden_license/bit-browser/src/popup/dirt/health/health-overview.component.ts [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+import { AtRiskGaugeComponent } from "../shared/at-risk-gauge/at-risk-gauge.component";
+
+import { RiskCategoryItemComponent } from "./risk-category-item.component";
+
+/**
+ * How each risk category renders, in the fixed order the overview shows them.
+ * The order is the highest-risk-wins priority the report service applies:
+ * Exposed, then Weak, then Reused.
+ */
+const RISK_CATEGORY_ROWS: readonly {
```
Comment (lastbestdev) [inline bitwarden_license/bit-browser/src/popup/dirt/health/health-overview.component.ts:31]: We likely need to update these with translation keys for handling empty states (i.e. 0 items found in category). See the following design screenshot with "No exposed passwords", "No weak passwords", ...

Image

File: bitwarden_license/bit-browser/src/popup/dirt/health/health-overview.component.html [hunk 5]
```diff
@@ -0,0 +1,44 @@
+@if (hasReport()) {
+  <div
+    class="tw-mx-4 tw-my-3 tw-flex tw-flex-col tw-items-center tw-rounded-xl tw-border tw-border-solid tw-border-border-base tw-bg-background tw-px-4 tw-pb-4 tw-pt-3"
+  >
```
Comment (lastbestdev) [inline bitwarden_license/bit-browser/src/popup/dirt/health/health-overview.component.html]: Can we use the `bit-card` component for the parent here instead of styled `div`? https://components.bitwarden.com/?path=/docs/component-library-cards-card--docs

File: apps/browser/src/_locales/en/messages.json [hunk 6]
```diff
@@ -7515,5 +7515,103 @@
...
+      "total_secure": {
+        "content": "$1",
+        "example": "47"
+      },
+      "total": {
+        "content": "$2",
+        "example": "47"
+      }
+    }
+  },
```
Comment (lastbestdev) [inline apps/browser/src/_locales/en/messages.json]: I believe the clean state (no at risk passwords) actually is supposed to reuse the same copy as the key above (`passwordsNeedFixing`). This can be removed

File: bitwarden_license/bit-browser/src/popup/dirt/health/health-overview.component.html [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    </h2>
+
+    <span bitTypography="helper" class="tw-text-center tw-text-fg-body-subtle">
+      @if (isAtRisk()) {
+        {{ "passwordsNeedFixing" | i18n: atRiskCount() : totalCount() }}
+      } @else {
+        <!-- Two placeholders, per the prototype copy: the healthy branch only
+             renders when at-risk is 0, so the secure count equals the total. -->
+        {{ "passwordsAreSecure" | i18n: totalCount() : totalCount() }}
+      }
```
Comment (lastbestdev) [inline bitwarden_license/bit-browser/src/popup/dirt/health/health-overview.component.html]: This copy should actually render the same regardless of clean/at risk state. See screenshot of clean scan I linked elsewhere in this review

```suggestion
     
        {{ "passwordsNeedFixing" | i18n: atRiskCount() : totalCount() }}

```

File: bitwarden_license/bit-browser/src/popup/dirt/health/risk-category-item.component.html [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+  <span slot="secondary">{{ descriptionKey() | i18n }}</span>
+
+  @if (isHealthy()) {
+    <bit-icon
+      slot="end"
+      name="bwi-check-circle"
+      class="tw-text-fg-success"
+      [ariaLabel]="'categoryHealthy' | i18n"
+    />
+  }
```
Comment (lastbestdev) [inline bitwarden_license/bit-browser/src/popup/dirt/health/risk-category-item.component.html]: Per design, we want the green check rendered within the tile at the beginning of the row (i.e. L2 of this component). This condition may be removed.

File: bitwarden_license/bit-browser/src/popup/dirt/health/health-overview.component.html [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+      <bit-item>
+        <dirt-risk-category-item
+          [labelKeySingular]="row.labelKeySingular"
+          [labelKeyPlural]="row.labelKeyPlural"
+          [descriptionKey]="row.descriptionKey"
+          [count]="row.count"
+          [icon]="row.icon"
+          [variant]="row.variant"
+          [route]="row.route"
+        />
```
Comment (lastbestdev) [inline bitwarden_license/bit-browser/src/popup/dirt/health/health-overview.component.html]: I wouldn't be opposed to rolling the `dirt-risk-category-item` component up into this one, since it is small, doesn't do much on its own, and isn't likely to be reused anywhere else. 

At the very least, I think it should be renamed to avoid confusion with the other item rows that are used in this feature (for displaying items in the at risk categories). Maybe including "Nav" or "Navigation" in the name could help?

File: bitwarden_license/bit-browser/src/popup/dirt/health/risk-category-item.component.ts [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+@Component({
+  selector: "dirt-risk-category-item",
+  templateUrl: "./risk-category-item.component.html",
+  // bit-item projects into <bit-item-action class="tw-flex tw-flex-1">, and
+  // bit-item-content relies on being that flex child: its host carries
+  // tw-w-full + tw-justify-between to push slot="end" to the row's edge.
+  // Without display:contents this wrapper becomes the flex item instead,
+  // shrinks to fit its text, and the chevron lands mid-row at a different
+  // position on every row, leaving the rest of the row unclickable.
+  host: { class: "tw-contents" },
```
Comment (lastbestdev) [inline bitwarden_license/bit-browser/src/popup/dirt/health/risk-category-item.component.ts]: Nit: this is a lot to read again, seems to document a failure/issue that came up during an implementation approach. I don't think it adds a lot of value and slows down readability for what is otherwise a very simple component

File: bitwarden_license/bit-browser/src/popup/dirt/health/risk-category-item.component.html [hunk 11]
```diff
@@ -0,0 +1,16 @@
+<a bit-item-content [routerLink]="route()" [truncate]="false">
+  <bit-icon-tile slot="start" [icon]="icon()" [variant]="variant()" />
+
+  {{ labelKey() | i18n: count() }}
```
Comment (lastbestdev) [inline bitwarden_license/bit-browser/src/popup/dirt/health/risk-category-item.component.html]: Thinking this should likely be wrapped in some sort of text element `p, h3, ...`

**Review by lastbestdev — COMMENTED** [review]
A few touch ups needed on the clean/no risks found state, but otherwise looking good.

**Comment by lastbestdev** [comment]
@AlexRubik, I'd keep an eye on Claude sticking comments in the code about problems that came up, or things it learned during development that could've been just relayed back to you during/after the CLI session and left there. Several of the nit comments on here relate to that

I also acknowledge that I personally air on the side of commenting less in code that I write, so take that with a grain of salt 😄

File: bitwarden_license/bit-browser/src/popup/dirt/health/risk-category-item.component.html (hunk 11 above)
Comment (AlexRubik) [reply bitwarden_license/bit-browser/src/popup/dirt/health/risk-category-item.component.html]: bit-item-content already wraps the default slot in a bitTypography="body2" tw-text-main div, so this renders inside a styled element rather than as a loose string. This is also the convention in most of the repo so I'm going to leave it as is for now.

File: bitwarden_license/bit-browser/src/popup/dirt/health/health-overview.component.ts (hunk 2 above)
Comment (AlexRubik) [reply bitwarden_license/bit-browser/src/popup/dirt/health/health-overview.component.ts]: https://github.com/bitwarden/clients/pull/22235/commits/de996d275d2fa3bd4c6c98e43d37b38a7dc5124b

File: bitwarden_license/bit-browser/src/popup/dirt/health/health-overview.component.ts (hunk 3 above)
Comment (AlexRubik) [reply bitwarden_license/bit-browser/src/popup/dirt/health/health-overview.component.ts]: https://github.com/bitwarden/clients/pull/22235/commits/de996d275d2fa3bd4c6c98e43d37b38a7dc5124b

File: bitwarden_license/bit-browser/src/popup/dirt/health/health-overview.component.ts (hunk 4 above)
Comment (AlexRubik) [reply bitwarden_license/bit-browser/src/popup/dirt/health/health-overview.component.ts:31]: https://github.com/bitwarden/clients/pull/22235/commits/28f156eb8a40877360e985bd22aac0bea1a81b7d

File: bitwarden_license/bit-browser/src/popup/dirt/health/health-overview.component.html (hunk 5 above)
Comment (AlexRubik) [reply bitwarden_license/bit-browser/src/popup/dirt/health/health-overview.component.html]: https://github.com/bitwarden/clients/pull/22235/commits/1111ae8f16109d23d9e0ebb971643b525de904b6

File: apps/browser/src/_locales/en/messages.json (hunk 6 above)
Comment (AlexRubik) [reply apps/browser/src/_locales/en/messages.json]: https://github.com/bitwarden/clients/pull/22235/commits/28f156eb8a40877360e985bd22aac0bea1a81b7d

File: bitwarden_license/bit-browser/src/popup/dirt/health/health-overview.component.html (hunk 7 above)
Comment (AlexRubik) [reply bitwarden_license/bit-browser/src/popup/dirt/health/health-overview.component.html]: https://github.com/bitwarden/clients/pull/22235/commits/28f156eb8a40877360e985bd22aac0bea1a81b7d

File: bitwarden_license/bit-browser/src/popup/dirt/health/risk-category-item.component.html (hunk 8 above)
Comment (AlexRubik) [reply bitwarden_license/bit-browser/src/popup/dirt/health/risk-category-item.component.html]: https://github.com/bitwarden/clients/pull/22235/commits/28f156eb8a40877360e985bd22aac0bea1a81b7d

File: bitwarden_license/bit-browser/src/popup/dirt/health/health-overview.component.html (hunk 9 above)
Comment (AlexRubik) [reply bitwarden_license/bit-browser/src/popup/dirt/health/health-overview.component.html]: https://github.com/bitwarden/clients/pull/22235/commits/b85887c2727afff44a90748417b9e30e89e65e06

File: bitwarden_license/bit-browser/src/popup/dirt/health/risk-category-item.component.ts (hunk 10 above)
Comment (AlexRubik) [reply bitwarden_license/bit-browser/src/popup/dirt/health/risk-category-item.component.ts]: https://github.com/bitwarden/clients/pull/22235/commits/de996d275d2fa3bd4c6c98e43d37b38a7dc5124b

File: bitwarden_license/bit-browser/src/popup/dirt/health/risk-category-nav-item.component.ts [hunk 12]
```diff
@@ -0,0 +1,99 @@
...
+ * A single row on the Health Overview: one vault-health risk category with its
+ * deduplicated at-risk count, navigating to that category's detail list. Named
+ * for the navigation so it stays distinct from the rows inside a category,
+ * which list individual logins.
+ *
+ * Presentational: it derives everything from its inputs and emits no events.
+ *
+ * Renders the `<a bit-item-content>` only. The surrounding `<bit-item>` is
+ * supplied by the consumer so that sibling rows are true siblings, which
+ * `bit-item-group`'s compact-mode corner rounding depends on.
```
Comment (Banrion) [inline bitwarden_license/bit-browser/src/popup/dirt/health/risk-category-nav-item.component.ts]: ⛏️ You can remove the presentational comment as well as the technical focused "what it renders". Comments should avoid repeating what code/the template explains clearly

File: bitwarden_license/bit-browser/src/popup/dirt/health/risk-category-nav-item.component.ts [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+  ItemModule,
+} from "@bitwarden/components";
+import { I18nPipe } from "@bitwarden/ui-common";
+
+/**
+ * A single row on the Health Overview: one vault-health risk category with its
+ * deduplicated at-risk count, navigating to that category's detail list. Named
+ * for the navigation so it stays distinct from the rows inside a category,
+ * which list individual logins.
+ */
```
Comment (lastbestdev) [inline bitwarden_license/bit-browser/src/popup/dirt/health/risk-category-nav-item.component.ts:18]: Nit: more comment bloat, the last detail about why the component is named how it is is unnecessary

File: apps/browser/src/_locales/en/messages.json [hunk 14]
```diff
@@ -7515,5 +7515,113 @@
...
+    "message": "No exposed passwords",
+    "description": "Title of the exposed risk category row when no passwords are exposed."
+  },
+  "exposedPassword": {
+    "message": "1 exposed password",
+    "description": "Title of the exposed risk category row when exactly one password is exposed."
+  },
+  "exposedPasswordsPlural": {
+    "message": "$COUNT$ exposed passwords",
+    "description": "Title of the exposed risk category row, including the count. Also used for a count of zero.",
```
[the remark this answers is not in the document]
Comment (AlexRubik) [reply apps/browser/src/_locales/en/messages.json]: https://github.com/bitwarden/clients/pull/22235/commits/c5fa09d1d087d34c4bfe83dda095f75870d2992d

**Review by lastbestdev — APPROVED** [review]
LGTM. Won't block based on content of comments, but I do personally think we're doing too much with them

File: bitwarden_license/bit-browser/src/popup/dirt/health/risk-category-item.component.html (hunk 11 above)
Comment (lastbestdev) [reply bitwarden_license/bit-browser/src/popup/dirt/health/risk-category-item.component.html]: I was wondering if it did something like this for us. Thank you for checking!
